js/ui/list.d.ts

```diff
@@ -31,6 +37,87 @@ import {
     SearchBoxMixinOptions
 } from './widget/ui.search_box_mixin';
 
+interface ListItemInfo {
+    readonly itemData?: any;
+    itemElement: TElement;
+    readonly itemIndex: number | { group: number; item: number; };
+}
+
+export interface ScrollInfo {
+    readonly scrollOffset?: any;
+    readonly reachedLeft: boolean;
+    readonly reachedRight: boolean;
+    readonly reachedTop: boolean;
+    readonly reachedBottom: boolean;
+}
+
+/** @public */
+export type ContentReadyEvent = ComponentEvent<dxList>;
+
+/** @public */
+export type DisposingEvent = ComponentDisposingEvent<dxList>;
+
+/** @public */
+export type GroupRenderedEvent = ComponentEvent<dxList> & {
+    readonly groupData?: any;
+    groupElement?: TElement;
+    readonly groupIndex?: number;
+}
+
+/** @public */
+export type InitializedEvent = ComponentInitializedEvent<dxList>;
+
+/** @public */
+export type ItemClickEvent = ComponentNativeEvent<dxList> & ListItemInfo;
+
+/** @public */
+export type ItemContextMenuEvent = ComponentNativeEvent<dxList> & ListItemInfo;
+
+/** @public */
+export type ItemDeletedEvent = ComponentEvent<dxList> & ListItemInfo;
+
+/** @public */
+export type ItemDeletingEvent = ComponentEvent<dxList> & ListItemInfo & {
+    cancel?: boolean | TPromise<void>;
+}
+
+/** @public */
+export type ItemHoldEvent = ComponentNativeEvent<dxList> & ListItemInfo;
+
+/** @public */
+export type ItemRenderedEvent = ComponentNativeEvent<dxList> & ItemInfo;
```

Comment:
@DokaRus. other list item events have  *grouped* index that can be `number | { group: number; item: number; }, but [ItemRendered](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxList/Configuration/#onItemRendered)
Is that correct?

Reply:
Yes, it looks inconsistent, but it reflects real behavior. `ItemRendered` in this field does not reflect whether the item belongs to a group.
I suggest we leave it as it is. I'll add the consistency discussion card for this event to our board and we'll fix it separately